### PR TITLE
msvc: preliminary support

### PR DESCRIPTION
**Pull request by lstipakov**
This adds msvc project files and tiny code changes,
required to build this project with msvc.

Microsoft resource compiler, unlike mingw's windres,
doesn't fully support macros and multi-line strings,
so I had to create a separate resource file for msvc.

Signed-off-by: Lev Stipakov <lev@openvpn.net>

### CODE REVIEW

**Comment by lstipakov** [comment]
ping @selvanair @mattock

File: res/openvpn-gui-res.rc [hunk 1]
```diff
@@ -58,6 +59,9 @@ ID_ICO_DISCONNECTED  ICON  DISCARDABLE  "disconnected.ico"
 #include "openvpn-gui-res-ua.rc"
 #include "openvpn-gui-res-zh-hans.rc"
 #include "openvpn-gui-res-zh-hant.rc"
+#else
+#include "openvpn-gui-res-en-msvc.rc"
+#endif
```
Comment (selvanair) [inline res/openvpn-gui-res.rc]: I think this would be cleaner if used as
```
#ifdef _MSC_VER
#include "openvpn-gui-res-en-msvc.rc"
#else 
<existing includes>
#endif 
```

File: main.c [hunk 2]
```diff
@@ -31,7 +31,9 @@
 #include <shlwapi.h>
 #include <wtsapi32.h>
 #include <prsht.h>
+#ifndef _MSC_VER
 #include <pbt.h>
```
Comment (selvanair) [inline main.c]: pbt.h is an obsolete header  -- I think we can just remove it. May be as a separate commit?

**Comment by selvanair** [comment]
@lstipakov This should be enough for a quick test build using MSVC. But, if we want to support MSVC builds by users (or for releases) we'll need a script that converts all resource files. Manual conversion is error-prone and hard to review. We cant possibly keep two sets of rc files in sync. 

With that caveat, I'm okay with this as a "preliminary" support. I haven't tested it, but if it works for you, good enough.

File: res/openvpn-gui-res.rc (hunk 1 above)
Comment (lstipakov) [reply res/openvpn-gui-res.rc]: agreed

File: main.c (hunk 2 above)
Comment (lstipakov) [reply main.c]: moved into separate commit

**Comment by lstipakov** [comment]
> @lstipakov This should be enough for a quick test build using MSVC. But, if we want to support MSVC builds by users (or for releases) we'll need a script that converts all resource files. Manual conversion is error-prone and hard to review. We cant possibly keep two sets of rc files in sync.

Well, it should be enough to develop with VS, except anything related to resource files - you can build, run, debug, edit code etc.

I've made a script which removes newlines and joins multiline LTEXT into single line. But VS is also not happy about using macros in LTEXT values, so there is still some work to do.

> With that caveat, I'm okay with this as a "preliminary" support.

Thanks, I have force pushed 2 commits with your suggestions. 

> I haven't tested it, but if it works for you, good enough.

I have also added appveyor script which builds VS project and saves artifact (openvpn-gui.exe).

**Comment by mattock** [comment]
@selvanair does "but if it works for you, good enough" mean we can merge this?

File: main.c [hunk 3]
```diff
@@ -31,7 +31,6 @@
 #include <shlwapi.h>
 #include <wtsapi32.h>
 #include <prsht.h>
-#include <pbt.h>
```
Comment (selvanair) [inline main.c]: Actually PBT macros are now in winuser.h which is included through windows.h. So even if we use them later, we dont need this obsolete header.
